content/blog/foundational-software/index.md

```diff
@@ -0,0 +1,221 @@
+
+> I've seen it play out as we built Aurora DSQL - we chose Rust for the new dataplane components, and started off developing other components with other tools. The control plane in Kotlin, operations tools in Typescript, etc. Standard "right tool for the job" stuff. But, as the team has become more and more familiar and comfortable with Rust, it's become the way everything is built. A lot of this is because we've seen the benefits of Rust, but at least some is because the team just enjoys writing Rust.
+>
+> -- [Marc Brooker, engineer at Amazon Web Services in Seattle on lobste.rs](https://smallcultfollowing.com/babysteps/blog/2025/05/15/10-years-of-rust/)
+
+That fully aligns with my experience: I find that teams become ambitious after a while.
+They reach for loftier goals because they can.
+The fact they don't have to deal with security issues anymore enables better affordances.
+From my conversations with other Rustaceans, we all made the same observation: suddenly we can build more ambitious projects that we never dared tackling before, such as writing a CPU emulator because we can.
+For fun.
+
+It feels to me as if this direction is more promising: starting with the foundational tech and growing into application-level/business-level code if needed/helpful.
+That's better than the other way around, which often feels unnecessarily clunky.
+Once the foundations are in Rust, other systems can be built on top of it.
+
+Just because we focus on foundational software doesn't mean we can't do other things.
+But the focus is to make sure that Rust stays true to its roots.
+
+## Systems That You Plan To Maintain For Years To Come
+
+So, what *is* foundational software?
+
+It's software that organizations deem critical for their success. 
+It might be:
+
+- a robust web backend
+- a package manager
+- a platform for robotics
+- a storage layer
+- a satellite control system 
+
+and many, many more.
+
+All of these things power organizations and *must not fail* or at least do so *gracefully*.
+My clients and the companies listed on our [podcast page](/podcast) all have one thing in common: 
+They work on Rust projects that are not on the sideline, but front and center, and they shape the future of their infrastructure.
+
+Rust is useful in situations where the ["worse is better" philosophy falls apart](https://dreamsongs.com/Files/worse-is-worse.pdf): it's a language for building the "right thing":
+
+> With the right thing, designers are equally concerned with simplicity, correctness, consistency, and completeness.
+
+I think many companies will choose Rust to build their future platforms on.
+As such, it competes with C++ as much as it does with Kotlin or Python.
+
+I believe that we should shift the focus away from memory safety (which these languages also have) and instead focus on the explicitness, expressiveness, and ecosystem of Rust that is highly competitive with these languages.
```

Comment:
You could also try to argue that this heavily depends on where people are coming from. If you try to market Rust as a good replacement for "systems languages" (read C), then the biggest selling feature is memory safety. If you try to convince people coming from interpreted languages than this will change. Also, for C devs who are only asking for a faster horse (a memory safe C), then the other benefits of Rust might resemble the shape of a Ford Model T.

Reply:
True. I could be more precise here, but I don't want to distract from my main argument and I believe it's a reasonable statement to make (in general and in the given context).